src/django_components/util/template_tag.py

```diff
@@ -155,287 +246,99 @@ def _parse_tag_preprocess(
     # Otherwise, depending on the tag spec, the tag may be:
     # 2. Block tag - With corresponding end tag, e.g. `{% endslot %}`
     # 3. Inlined tag - Without the end tag.
-    last_token = attrs[-1].serialize(omit_key=True) if len(attrs) else None
-
-    if last_token == "/":
+    last_token = attrs[-1].value if len(attrs) else None
+    if last_token and last_token.serialize() == "/":
         attrs.pop()
         is_inline = True
     else:
         is_inline = not tag_spec.end_tag
 
-    raw_args, raw_kwargs, raw_flags = _parse_tag_input(tag_name, attrs)
+    raw_params, flags = _extract_flags(tag_name, attrs, tag_spec.flags or [])
 
-    return tag_name, raw_args, raw_kwargs, raw_flags, is_inline
+    def _parse_tag_body(parser: Parser, end_tag: str, inline: bool) -> NodeList:
+        if inline:
+            body = NodeList()
+        else:
+            body = parser.parse(parse_until=[end_tag])
+            parser.delete_first_token()
+        return body
 
+    return ParsedTag(
+        tag_name=tag_name,
+        params=TagParams(params=raw_params, tag_spec=tag_spec),
+        flags=flags,
+        # NOTE: We defer parsing of the body, so we have the chance to call the tracing
+        # loggers before the parsing. This is because, if the body contains any other
+        # tags, it will trigger their tag handlers. So the code called AFTER
+        # `parse_body()` is already after all the nested tags were processed.
+        parse_body=lambda: _parse_tag_body(parser, tag_spec.end_tag, is_inline) if tag_spec.end_tag else NodeList(),
+    )
 
-def _parse_tag_input(tag_name: str, attrs: List[TagAttr]) -> Tuple[List[str], List[TagKwarg], Set[str]]:
```

Comment:
Lots of changes and deletions in this file, as previously we tried to extract and separate args and kwargs BEFORE resolving the values, while now we do it only at render time. Only flags (e.g. `{% slot "name" default %}`) are still extracted at the time when the template is being parsed / compiled.